### PR TITLE
Unit Tests for Enhanced Program Management Tool Doctype Module

### PR DESCRIPTION
**Pull request by SymonMuchemi**
This pull request introduces comprehensive test coverage for the `Enhanced Program Enrollment Tool` in `nl_school/junior_school_customization`. It includes unit tests for various scenarios, ensuring robust validation, correct functionality, and proper handling of edge cases.

#### Core Functionality Testing:
* Added `test_get_students_from_applicant` to validate that students are correctly retrieved from approved `Student Applicants` based on the specified program, academic year, and term.
* Added `test_get_students_from_program_enrollment` to ensure students are retrieved accurately from `Program Enrollment` records for a given academic year and term.

#### Validation and Error Handling:
* Added `test_get_students_returns_error_when_no_students_found` to confirm an error is raised when no students match the specified criteria.
* Added `test_get_students_throws_if_academic_year_missing` and `test_get_students_throws_if_program_missing_for_applicant` to validate that mandatory fields (`Academic Year` and `Program`) are enforced.
* Added `test_get_students_with_invalid_source_throws_no_students` to ensure an error is raised when an invalid source is provided.

#### Edge Case Handling
This pull request enhances the test coverage for the `Enhanced Program Enrollment Tool` by adding multiple test cases to validate its functionality. The changes include setting up and tearing down test data, testing various scenarios for retrieving students, and verifying error handling and workflows.

### Test Coverage Enhancements:

* **Setup and Teardown for Test Data**:
  - Added `setUp` and `tearDown` methods to create and clean up test data for `Student Applicant`, `Student`, and `Program Enrollment` records. This ensures consistent test environments.

* **New Test Cases for `get_students` Method**:
  - Added tests to validate that students are correctly retrieved from `Student Applicant` and `Program Enrollment` based on specified criteria.
  - Implemented a test to ensure disabled students are filtered out when retrieving students from `Program Enrollment`.
  - Added tests to verify that `get_students` raises appropriate errors when no students are found, or when mandatory fields like `academic_year` or `program` are missing.
  - Included a test to handle invalid sources for the `get_students_from` field, ensuring proper error messages are raised.

* **Workflow Validation for `enroll_students` Method
This pull request adds extensive test coverage for the `Enhanced Program Enrollment Tool` in the `nl_school` module. The changes include new test cases to validate the tool's functionality, error handling, and workflows. Additionally, the `setUp` and `tearDown` methods are implemented to manage test data setup and cleanup.

### New Test Cases for `Enhanced Program Enrollment Tool`:

* **Basic Functionality:**
  - Added `test_get_students_from_applicant` to verify that students are correctly retrieved from approved `Student Applicant` records.
  - Added `test_get_students_from_program_enrollment` to validate that students are retrieved from `Program Enrollment` records for a given academic year and term.

* **Error Handling:**
  - Added `test_get_students_returns_error_when_no_students_found` to confirm that an error is raised when no students match the specified criteria.
  - Added `test_get_students_throws_if_academic_year_missing` and `test_get_students_throws_if_program_missing_for_applicant` to ensure that mandatory fields (`Academic Year` and `Program`) are validated.
  - Added `test_get_students_with_invalid_source_throws_no_students` to validate error handling for invalid data sources.

### CODE REVIEW

File: nl_school/junior_school_customization/doctype/enhanced_program_enrollment_tool/test_enhanced_program_enrollment_tool.py [hunk 1]
```diff
@@ -1,9 +1,271 @@
...
+                "student_category": "General",
+                "program": self.program,
+                "company": "Test Company",
+                "custom_stream": "Stream A",
+                "academic_year": self.academic_year,
+                "academic_term": self.academic_term,
+            }
+        ).insert(ignore_permissions=True)
+
+    def tearDown(self):
```
Comment (maniamartial) [inline nl_school/junior_school_customization/doctype/enhanced_program_enrollment_tool/test_enhanced_program_enrollment_tool.py:52]: After teardown(), which apparently will delete the defined doctype, commit because it's the last function being called.
frappe.db.commit()

File: nl_school/junior_school_customization/doctype/enhanced_program_enrollment_tool/test_enhanced_program_enrollment_tool.py [hunk 2]
```diff
@@ -1,9 +1,271 @@
...
 class TestEnhancedProgramEnrollmentTool(FrappeTestCase):
-    pass
+    def setUp(self):
+        # test data for Student Applicant
+        self.academic_year = "2025-2026"
+        self.program = "Test Program"
+        self.academic_term = "Term 1"
+
+        # Insert a Student Applicant
+        self.student_applicant = frappe.get_doc(
```
Comment (maniamartial) [inline nl_school/junior_school_customization/doctype/enhanced_program_enrollment_tool/test_enhanced_program_enrollment_tool.py]: Before inserting/creating the test doc, please make sure you validate that they don't exist.
`if frappe.db.exists("Student Applicant", {values}, cache=False):
            return.`

**Review by maniamartial — CHANGES_REQUESTED** [review]

File: nl_school/junior_school_customization/doctype/enhanced_program_enrollment_tool/test_enhanced_program_enrollment_tool.py (hunk 2 above)
Comment (SymonMuchemi) [reply nl_school/junior_school_customization/doctype/enhanced_program_enrollment_tool/test_enhanced_program_enrollment_tool.py]: Suggested changes in commits [091946c](https://github.com/navariltd/Junior-School/pull/73/commits/091946c5071762e7b6496e208b69f9a81dabfddc) and [1d49981](https://github.com/navariltd/Junior-School/pull/73/commits/1d49981ad07a66be3aeebb7ad9bf413fde68fe15)

File: nl_school/junior_school_customization/doctype/enhanced_program_enrollment_tool/test_enhanced_program_enrollment_tool.py (hunk 1 above)
Comment (SymonMuchemi) [reply nl_school/junior_school_customization/doctype/enhanced_program_enrollment_tool/test_enhanced_program_enrollment_tool.py:52]: Suggested change in this commit [00f29e7](https://github.com/navariltd/Junior-School/pull/73/commits/00f29e783cddf6772f8d6f21bc91823a29068d1d)
